packages/store/src/internal/state-stream.ts

```diff
@@ -3,13 +3,19 @@ import { BehaviorSubject } from 'rxjs';
 
 import { PlainObject } from '@ngxs/store/internals';
 
+import { InternalNgxsExecutionStrategy } from '../execution/internal-ngxs-execution-strategy';
+
 /**
  * BehaviorSubject of the entire state.
  * @ignore
  */
 @Injectable()
 export class StateStream extends BehaviorSubject<PlainObject> {
-  constructor() {
+  constructor(private _ngxsExecutionStrategy: InternalNgxsExecutionStrategy) {
     super({});
   }
+
+  next(value: PlainObject): void {
+    this._ngxsExecutionStrategy.leave(() => super.next(value));
+  }
```

Comment:
I think that this could have unintended side effects for other aspects of the state management (like plugins etc.).
I think it would be a better approach if `Store` had a derived state stream that calls the `leave` to change the execution context, and then is shared using the `shareReplay` operator.
All selects would then use this, and it would call `leave` only once for any state change across all active selectors. What do you think @arturovt?
This would be used by the `select` and `subscribe` methods of the `Store`.

Reply:
@markwhitfeld I've made some changes, could you have a look. This is the only approach that doesn't break unit tests, using `shareReplay` breaks a lot of tests, I didn't dig why 😄